src/mscorlib/src/System/Text/UTF8Encoding.cs

```diff
@@ -53,9 +53,16 @@ public class UTF8Encoding : Encoding
 
         private const int UTF8_CODEPAGE = 65001;
 
+        // Allow for devirtualization (see https://github.com/dotnet/coreclr/pull/9230)
+        [Serializable]
+        internal sealed class UTF8EncodingSealed : UTF8Encoding
```

Comment:
Its exposed via Encoding as 
```
Encoding UTF8 => UTF8Encoding.s_default;
```
so it needs to be able to see the class; else `s_default` being that type will be a compile error; if the `s_default` field was `UTF8Encoding` it might be too much of a stretch to infer it was actually `UTF8EncodingSealed` as the type would have to store what it had actually been initialised to?

Don't know if the devirtualization extends to inspecting what a readonly static actually is?